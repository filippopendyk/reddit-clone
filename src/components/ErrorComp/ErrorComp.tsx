type Props = {
    error: string;
}

const ErrorComp: React.FC<Props> = ({error}) => {
    return <>
        <h4>Error while loading posts! Try again!</h4>
        <p>Reason: {error}</p>
    </>
}

export default ErrorComp;