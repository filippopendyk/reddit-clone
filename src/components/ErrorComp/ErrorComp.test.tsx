import ErrorComp from "./ErrorComp";
import renderer from 'react-test-renderer';
import { render, screen } from "@testing-library/react";

describe('ErrorComp', () => {
    it('Renders without crashing', () => {
        const tree = renderer
        .create(
            <ErrorComp error={'network error'}/>
        )
        .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Renders the passed error string', () => {
        const exampleError = 'network error';

        render(
            <ErrorComp error={exampleError}/>
        );

        const error = screen.getByText(`Reason: ${exampleError}`);
        expect(error).toBeInTheDocument();
    })
})