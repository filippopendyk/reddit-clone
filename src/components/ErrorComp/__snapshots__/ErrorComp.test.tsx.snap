// Vitest Snapshot v1

exports[`ErrorComp > Renders without crashing 1`] = `
[
  <h3>
    Error while loading posts! Try again!
  </h3>,
  <p>
    Reason: 
    network error
  </p>,
]
`;
